### PR TITLE
refactoring: OrderFeeProcessor and improved logging

### PR DESCRIPTION
**Pull request by vladimir-8**
- refactoring: OrderFeeProcessor: update adjustments' values instead of removing and re-adding them to avoid https://github.com/coopcycle/coopcycle-web/issues/4128

- improved logging

### CODE REVIEW

File: src/Sylius/OrderProcessing/OrderFeeProcessor.php [hunk 1]
```diff
@@ -146,39 +136,29 @@ public function process(BaseOrderInterface $order): void
...
-                    $order->addAdjustment($tipAdjustment);
-
-                    $this->logger->info(sprintf('Order %s | OrderFeeProcessor | tip amount: %d | re-used previous tip adjustment',
-                        $this->loggingUtils->getOrderId($order), $tipAdjustment->getAmount()));
-                }
-            }
+        $newTipAmount = $order->getTipAmount();
+
+        // $order->getTipAmount() is the tip amount set in the current request
+        // preserve the previous tip amount if it's null
```
Comment (Atala) [inline src/Sylius/OrderProcessing/OrderFeeProcessor.php:142]: "if the new tip amount is null/not set" 

("it" could mean "the old tip amount")

File: src/Sylius/OrderProcessing/OrderFeeProcessor.php [hunk 2]
```diff
@@ -263,4 +236,48 @@ private function getDelivery(OrderInterface $order)
 
         return $delivery;
     }
+
+    private function lazyUpsertAdjustment(
```
Comment (Atala) [inline src/Sylius/OrderProcessing/OrderFeeProcessor.php:240]: why is it `lazy` ?

File: src/Controller/StripeController.php [hunk 3]
```diff
@@ -464,6 +465,7 @@ public function createPaymentIntentAction($hashId, Request $request,
             ->find($paymentId);
 
         if (null === $payment) {
+            $this->logger->warning(sprintf('Payment with id "%d" does not exist', $paymentId), ['hash' => $hashId]);
```
Comment (Atala) [inline src/Controller/StripeController.php:468]: maybe log as error ?

File: src/Sylius/OrderProcessing/OrderFeeProcessor.php (hunk 2 above)
Comment (vladimir-8) [reply src/Sylius/OrderProcessing/OrderFeeProcessor.php:240]: Indeed, now it's not lazy. The first version was. I'll rename it

**Comment by vladimir-8** [comment]
@Atala in the last few commits I added: 

- a unique `request_id`, so that we can distinguish log entries added by concurrent processes ( https://coopcycle.slack.com/archives/G855MRP7V/p1714150890342309?thread_ts=1713919143.563899&cid=G855MRP7V )

- `method`, `request_uri` and route (controller) to provide more context while debugging

```
{
  "message": "Order #713 | CheckoutSubscriber | postFlush | inserted: Sylius\\Component\\Order\\Model\\OrderItemUnit id:2057",
  "context": {},
  "level": 200,
  "level_name": "INFO",
  "channel": "checkout",
  "datetime": "2024-04-29T17:07:21.108271-07:00",
  "extra": {
    "token": {
      "authenticated": true,
      "roles": [
        "ROLE_USER"
      ],
      "user_identifier": "user_1",
      "username": "user_1"
    },
    "request_id": "beaa721e-b3f8-4f50-b071-445575028cc5",
    "method": "POST",
    "request_uri": "/api/orders/713/items",
    "requests": [
      {
        "controller": "AppBundle\\Action\\Cart\\AddItem",
        "route": "api_orders_post_cart_items_item",
        "route_params": {
          "_format": null,
          "_stateless": null,
          "_api_resource_class": "AppBundle\\Entity\\Sylius\\Order",
          "_api_identifiers": [
            "id"
          ],
          "_api_has_composite_identifier": false,
          "_api_item_operation_name": "post_cart_items",
          "id": "713"
        }
      }
    ]
  }
}
```

**Comment by Atala** [comment]
cool :+1: I don't know why tests are failing?

We can roll this out once we have updated the server.

**Pull request merged**